### ISSUE TITLE
QExpansionItem @aftershow and @afterhide

### ISSUE THREAD

**Issue by labs20** [quasarframework/quasar#5664]
**Is your feature request related to a problem? Please describe.**
If you need to code something that **must** happen only after the component is **fully** expanded or closed, right now you cant.

**Describe the solution you'd like**
The QExpansionItem should emit an "aftershow" and an "afterhide" event at the end of the respective animations.

**Describe alternatives you've considered**
Coding with a delay trying to compensate the animation time, but it gets very inconsistent results based on the current machine processing load and even from different machine setups.

**Additional context**
As a practical example, I'm trying to get a snapshot of the contents of a QExpansionItem as the user navigate among a few of them, and more often than expected, getting halfway snapshots.

**Comment by rfox12** [comment]
Those events would be useful for me as well (on any Quasar elements that can have a transition built in!).  At this point I do less with animated components specifically because there are no such events.  In the past I had code like this:
```
function whichTransitionEvent() {
  var t,
    el = document.createElement("fakeelement");

  var transitions = {
    "transition": "transitionend",
    "OTransition": "oTransitionEnd",
    "MozTransition": "transitionend",
    "WebkitTransition": "webkitTransitionEnd"
  }

  for (t in transitions) {
    if (el.style[t] !== undefined) {
      return transitions[t];
    }
  }
}
```
later in some initialization area:
```
mounted(){
        var transitionEvent = whichTransitionEvent();
        var animatedDiv = document.querySelector("...");
        animatedDiv .addEventListener(transitionEvent, (event) => {
            if (event.propertyName === 'width' && animatedDiv === event.target) {
              // your CSS animation is done ... do something
            }
          });
}
```
So basically I had to know what was animating (in this case "width" and have a reference to a mounted component "animatedDiv")

**Comment by rstoenescu** [comment]
New events will be available in "quasar" v1.9.0
